### ISSUE TITLE
How to use downloaded pretrained models

### ISSUE THREAD

**Issue by fatualux** [skyfireitdiy/codegeex-vim#1]
First of all, thank you for your work.
Is there a way to use my locally downloaded models instead of api keys and URL?
Just like I do in Code OSS, with VS Code Extension.

This is my Codegeex config:

# CodeGeeX-13B configuration

CHECKPOINT_PATH="/~/.cache/codegeex/codegeex_13b.pt"

MODEL_ARGS="--num-layers 39 \
            --hidden-size 5120 \
            --num-attention-heads 40 \
            --max-position-embeddings 2048 \
            --attention-softmax-in-fp32 \
            --load "$CHECKPOINT_PATH" \
            --layernorm-epsilon 1e-5 \
            --fp16 \
            --ws-encoding-start-id 10 \
            --ws-encoding-length 10 \
            --make-vocab-size-divisible-by 52224 \
            --seq-length 2048"

I tried to edit the python scrpts in \autoload\  by changing URL variable with the path where the .pt file is located.
In the status bar I see "CodeGeex is thinking..." but no results, and translating opens the list of languages supported but when i inset the number it returns an error.

Can you help me?

Thank you so much

Thank you

OS:
Linux
6.2.8-arch1-1

Vim Editor Version:

VIM - Vi IMproved 9.0 (2022 Jun 28, compiled Mar 21 2023 20:51:42)
Included patches: 1-1420
Compiled by Arch Linux

**Comment by fatualux** [comment]
![image](https://user-images.githubusercontent.com/35587292/231239089-b03f3a47-c1e4-4c58-b3e2-e807b60d222c.png)

**Comment by skyfireitdiy** [comment]
I'm very sorry, I'm too busy with work and I haven't checked my email for a while.
You need to debug on the local command line and modify the command line in the codegeex.vim file instead of modifying the Python script.
I haven't deployed it locally, so I don't know how to use it locally. Good luck to you.
Another way is to try other Codegeex vim plugins on Github, and I think there should be another one in my mind.

On Wednesday 12 April 2023 00:54:20 (+08:00), FZ wrote:



First of all, thank you for your work.
Is there a way to use my locally downloaded models instead of api keys and URL?
Just like I do in Code OSS, with VS Code Extension.

This is my Codegeex config:

CodeGeeX-13B configuration

CHECKPOINT_PATH="/~/.cache/codegeex/codegeex_13b.pt"

MODEL_ARGS="--num-layers 39
--hidden-size 5120
--num-attention-heads 40
--max-position-embeddings 2048
--attention-softmax-in-fp32
--load "$CHECKPOINT_PATH"
--layernorm-epsilon 1e-5
--fp16
--ws-encoding-start-id 10
--ws-encoding-length 10
--make-vocab-size-divisible-by 52224
--seq-length 2048"

I tried to edit the python scrpts in \autoload\ by changing URL variable with the path where the .pt file is located.
In the status bar I see "CodeGeex is thinking..." but no results, and translating opens the list of languages supported but when i inset the number it returns an error.

Can you help me?

Thank you so much

Thank you

OS:
Linux
6.2.8-arch1-1

Vim Editor Version:

VIM - Vi IMproved 9.0 (2022 Jun 28, compiled Mar 21 2023 20:51:42)
Included patches: 1-1420
Compiled by Arch Linux

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you are subscribed to this thread.


-- 
skyfire

**Comment by fatualux** [comment]
Ok, thanks so much.
